src/main/java/net/sf/jabref/importer/fileformat/BibTeXMLImporter.java

```diff
@@ -78,33 +98,177 @@ public ParserResult importDatabase(BufferedReader reader) throws IOException {
+                }
+                if (entry.getId() != null) {
+                    bibEntry.setCiteKey(entry.getId());
+                }
+                bibEntry.setField(fields);
+                bibItems.add(bibEntry);
+            }
+        } catch (JAXBException | XMLStreamException e) {
             LOGGER.error("Error with XML parser configuration", e);
             return ParserResult.fromErrorMessage(e.getLocalizedMessage());
-        } catch (org.xml.sax.SAXException e) {
-            LOGGER.error("Error during XML parsing", e);
-            return ParserResult.fromErrorMessage(e.getLocalizedMessage());
-        } catch (IOException e) {
-            LOGGER.error("Error during file import", e);
-            return ParserResult.fromErrorMessage(e.getLocalizedMessage());
         }
         return new ParserResult(bibItems);
     }
 
+    /**
+     * In this method, all <Code>get</Code> methods that t has will be used and their value will be put to fields,
+     * if it is not null. So for example if entryType has the method <Code>getAbstract</Code>, then
+     * "abstract" will be put as key to fields and the value of <Code>getAbstract</Code> will be put as value to fields.
+     * Some <Code>get</Code> methods shouldn't be mapped to fields, so <Code>getClass</Code> for example will be skipped.
+     *
+     * @param entryType This can be all possible BibTeX types. It contains all fields of the entry and their values.
+     * @param fields A map where the name and the value of all fields, that the entry contains, will be put.
+     */
+    private <T> void parse(T entryType, Map<String, String> fields) {
+        Method[] declaredMethods = entryType.getClass().getDeclaredMethods();
+        for (Method method : declaredMethods) {
+            try {
+                if (method.getName().equals("getYear")) {
+                    putYear(fields, (XMLGregorianCalendar) method.invoke(entryType));
+                    continue;
+                } else if (method.getName().equals("getNumber")) {
+                    putNumber(fields, (BigInteger) method.invoke(entryType));
+                    continue;
+                } else if (isFieldToSkip(method.getName())) {
+                    continue;
+                } else if (method.getName().contains("get")) {
+                    putIfValueNotNull(fields, method.getName().replace("get", ""), (String) method.invoke(entryType));
+                }
+            } catch (IllegalArgumentException | InvocationTargetException | IllegalAccessException e) {
+                LOGGER.error("Could not invoke method", e);
+            }
+        }
+    }
+
+    /**
+     * Returns whether the value of the given method name should be mapped or whether the field can be skipped.
+     *
+     * @param name of a method
+     * @return true if the field can be skipped, else false
+     */
+    private boolean isFieldToSkip(String name) {
+        return FIELDS_TO_SKIP.contains(name);
```

Comment:
Either remove the `get` prefix and check for ignored case or name this "ignoredMethods" etc. --> `getSize` is not a bibtex field.